pkgs/by-name/zi/zipline/package.nix

```diff
@@ -0,0 +1,148 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nodejs,
+  yarn-berry,
+  pkg-config,
+  makeWrapper,
+  cacert,
+  prisma-engines,
+  openssl,
+  ffmpeg,
+  python3,
+  vips,
+  nixosTests,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "zipline";
+  version = "3.7.11";
+
+  src = fetchFromGitHub {
+    owner = "diced";
+    repo = "zipline";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-sogsPx6vh+1+ew9o3/0B4yU9I/Gllo9XLJqvMvGZ89Q=";
+  };
+
+  patches = [
+    # Update prisma to match the version in nixpkgs exactly (currently 6.0.1). To create this patch, change the
+    # versions in `package.json`, then run `nix run nixpkgs#yarn-berry -- install --mode update-lockfile`
+    # to update `yarn.lock`.
+    ./prisma6.patch
+  ];
+
+  # we cannot use fetchYarnDeps because that doesn't support yarn 2/berry lockfiles
+  yarnOfflineCache = stdenv.mkDerivation {
+    pname = "yarn-deps";
+    inherit (finalAttrs) version src patches;
+
+    nativeBuildInputs = [ yarn-berry ];
+
+    dontInstall = true;
+
+    YARN_ENABLE_TELEMETRY = "0";
+    NODE_EXTRA_CA_CERTS = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+
+    configurePhase = ''
+      export HOME="$NIX_BUILD_TOP"
+      yarn config set enableGlobalCache false
+      yarn config set cacheFolder $out
+      yarn config set --json supportedArchitectures.os '[ "linux" ]'
+      yarn config set --json supportedArchitectures.cpu '[ "arm64", "x64" ]'
+    '';
+
+    buildPhase = ''
+      mkdir -p $out
+      yarn install --immutable --mode skip-build
+    '';
+
+    outputHash = "sha256-kWE6YVhyH5Lk/SO0h624Zq9/6ztoUE3FNzHB0dyl5aI=";
+    outputHashMode = "recursive";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    makeWrapper
+    nodejs
+    yarn-berry
+    python3
+  ];
+
+  buildInputs = [
+    vips
+    openssl
+  ];
+
+  NEXT_TELEMETRY_DISABLED = "1";
+  YARN_ENABLE_TELEMETRY = "0";
+
+  FFMPEG_BIN = lib.getExe ffmpeg;
+  PRISMA_SCHEMA_ENGINE_BINARY = lib.getExe' prisma-engines "schema-engine";
```

Comment:
Would be cool if we could dedupe those with the ones below somehow

Reply:
I moved them into an `environment` attrset which is used in both the configure phase and the wrapper script. Regarding your other review comment, escaping these values is probably not necessary right now, however in the future one might need to add more environment variables, so I figured it would be a good idea to add the escaping now, so we don't have to rely on `environment` to only contain already escaped values.